### ISSUE TITLE
Merge work into new org?

### ISSUE THREAD

**Issue by tiesont** [lddubeau/bootprompt#111]
Hello there;

I'm in the process of shifting my focus from trying to maintain Bootbox to creating a new project in an organization (so that there's no "what happens when the owner disappears" problem again). Do you have any interest in joining https://github.com/BootboxJS and merging your work there? You've done most of what I've been wanting to do for a while, in your Typescript version, so it'd be great not to have to revisit work you've already completed.

If not, no offense taken.

**Comment by lddubeau** [comment]
I agree. It would be best for everyone to put my changes under an organization, and hopefully have more than just me be responsible for maintaining it.

What form of "merge" are you thinking about? Trying to use git to merge the master branch of bootbox post-fork with the master branch of bootprompt is likely to be extremely time consuming, if not downright unmanageable. Another approach might be to take the master branch of bootprompt as the starting point for new development, but that means being willing to live with [breaking changes](https://github.com/lddubeau/bootprompt/blob/master/docs/bootbox-to-bootprompt.md).

**Comment by tiesont** [comment]
Unless we really wanted to preserve the history of the Bootbox codebase, I think either starting with your master branch might make the most sense. The alternative is to cherry-pick what you think you really got right, and then fill in from there.

One thing I thought might be worth exploring is breaking the codebase into modules (I'm using the term loosely, since I know some frameworks have a specific "thing" that you can call a "module"). 

For example, building a core module that contains only the code to create dialogs (roughly the functionality encapsulated in `bootbox.dialog(...)` at the moment). So, if all you (the developer) wanted was something to make your custom modals easier to use, you could just import the dialog module. Then, we'd build alert, confirm, and prompt on top of that (basically, what already happens, but probably architected better). I'd also like to improve how locales are loaded, which (with a switch to Typescript and an actual build script) I can see making for a better ability to customize what's available.

There's more to go over, but if (at a minimum) the two of us could combine our efforts on one project, I think I'd feel better about keeping some version of Bootbox going. Which... segues into one last point: the name. I went with BootboxJS because (a) Bootbox is a name that resonates with me, (b) bootboxjs.org and bootboxjs.net were both available as domains, and (c) BB makes a nifty enough logo (you can see my first rough version on the org page). If you prefer something else, we can always rename the organization.

I'll send you an invite to the org, which you can accept or decline as you see fit. Regardless, thank you for taking some time to respond.

**Comment by tiesont** [comment]
Meet or Teams would work. Google Chat works too - depends on what kind of meeting (chat/voice/video?) everyone is interested in.

@lddubeau I don't think either @tarlepp or I are in any kind of hurry to do anything, so just let us know when you're ready.

**Comment by lddubeau** [comment]
I'm still here, and unfortunately still in emergency mode. My time is likely to be substantially tied up for the foreseeable future. But I do want the migration to happen! And I do want to help make it happen. And I don't want to tie everything up indefinitely.

So here is what I would suggest. I would have preferred having a meeting but I'm in a situation where finding a block of time safe from interruptions is not doable. BootboxJS could have a pseudo-repo under it which is created for the sake of discussing the overall plan. It would have no code but each item on the agenda of the meeting we would have had could be an issue filed in the pseudo-repo. (Or maybe github has some better functionality now. I've not kept track of all the new features.)

This would not have been my first choice, but a way to make the best of a bad situation.

I have no preference regarding the name of the organization.

I'll leave the technical discussion of modules, and so on, for whatever venue we decide to use to hold our discussion.

**Comment by tiesont** [comment]
Sorry, work's been kind of nuts all summer (suppose it's better than being unemployed). I haven't really been able to do much to move this forward.
